dbmongo/js/compact/reduce.ts

```diff
@@ -105,14 +98,12 @@ export function reduce(
           })
         }
       } else {
-        Object.keys(
-          reduced_value.batch[batch][type as keyof BatchValue] || {}
-        ).forEach((hash) => {
+        for (const hash in currentBatch[type as keyof BatchValue]) {
```

Comment:
@JazzyPierrot pour info, l'usage d'une boucle `for` ici permet effectivement de se passer du `|| {}` (objet vide par défaut, au cas où `currentBatch[type] == undefined`), car, contrairement à `Object.keys`, la boucle `for` tolère la présence de `undefined` comme domaine itérable. (dans ce cas, la boucle ne s'exécute pas)

Ceci dit, je compte quand même voir s'il existe un moyen d'exprimer à TS que chaque prop de `BatchValue` ne peut être `undefined` si elle est fournie dans l'objet.

Reply:
Ok, ça fait le taf !